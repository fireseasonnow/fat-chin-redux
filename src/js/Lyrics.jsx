import React from 'react';
import styled from 'styled-components';

const Lyrics = ({
    lyricsVisibilityHandler,
    name,
    text,
}) => (
    <Inner>
        <Close onClick={() => lyricsVisibilityHandler()} type='button' />
        <Text>
            <Heading>{name}</Heading>
            {text || 'Instrumental'}
        </Text>
    </Inner>
);

export default Lyrics;

const Inner = styled.div`
    display: grid;
    grid-template-columns: 1fr auto 1fr;
    height: 100%;
    overflow-y: scroll;
`;

const Close = styled.button`
    align-self: start;
    top: 42px;
    align-self: center;
    justify-self: end;
    background: none;
    color: #fff;
    border: 0;
    position: relative;
    left: -15px;
    width: 40px;
    height: 40px;
    padding-right: 28px;

    &:hover {
        cursor: pointer;
    }

    &:before {
        content: '';
        position: absolute;
        top: 20px;
        width: 21px;
        height: 1px;
        background-color: currentColor;
        transform: rotate(-45deg);
    }

    &:after {
        content: '';
        position: absolute;
        top: 20px;
        width: 21px;
        height: 1px;
        background-color: currentColor;
        transform: rotate(45deg);
    }
`;

const Text = styled.p`
    margin: 50px 0;
    white-space: pre-line;
`;

const Heading = styled.span`
    display: block;
    text-transform: uppercase;
    margin-bottom: 20px;
`;
