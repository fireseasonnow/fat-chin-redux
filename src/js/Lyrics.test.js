import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';

import Lyrics from './Lyrics';

configure({ adapter: new Adapter() });

describe('<Lyrics />', () => {
    let wrapper = {};

    beforeEach(() => {
        wrapper = shallow(<Lyrics />);
    });

    it('should render without crashing', () => {
        expect(wrapper).toBeTruthy();
    });

    it('should render lyrics with name and text', () => {
        wrapper.setProps({
            name: 'Hello',
            text: 'Lorem Ipsum',
        });

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('when lyrics are missing should render lyrics with name and "Instrumental" text', () => {
        wrapper.setProps({
            name: 'Hello',
            text: '',
        });

        const check = wrapper.find('Lyrics__Text').contains('Instrumental');

        expect(check).toBeTruthy();
    });

    it('should fire lyricsVisibilityHandler when clicked on a button', () => {
        const lyricsVisibilityHandler = jest.fn();

        wrapper.setProps({
            lyricsVisibilityHandler,
        });

        wrapper.find('Lyrics__Close').simulate('click');

        expect(lyricsVisibilityHandler).toHaveBeenCalled();
    });
});
