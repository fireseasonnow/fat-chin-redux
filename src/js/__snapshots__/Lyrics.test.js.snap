// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lyrics /> should render lyrics with name and text 1`] = `
<Lyrics__Inner>
  <Lyrics__Close
    onClick={[Function]}
    type="button"
  />
  <Lyrics__Text>
    <Lyrics__Heading>
      Hello
    </Lyrics__Heading>
    Lorem Ipsum
  </Lyrics__Text>
</Lyrics__Inner>
`;
